packages/webdriverio/src/commands/browser/url.js

```diff
@@ -35,9 +37,29 @@
 import nodeUrl from 'url'
 import { validateUrl } from '../../utils'
 
-export default function url (path) {
+export default async function url (path, { inject } = {} ) {
+    /**
+     * If inject is a function, then we attach puppeteer to the existing browser instance and
+     * injects the passed function on the active tab
+     *
+     */
+    if (inject) {
+        if (typeof inject !== 'function') {
+            throw new Error('Parameter "inject" for url command needs to be type of function')
+        }
+        const browser = await this.getPuppeteer()
```

Comment:
Assume puppeteer should be attached seems pretty limiting for this feature. Are we sure this assumption is desirable?

For example, one might want to inject axe-core onto the page for accessibility testing, but doesn't use puppeteer.

Reply:
If attaching Puppeteer is the desired default behavior, would it make sense to create an opt-out option ?

Reply:
This doesn't attach Puppeteer to anything. We need Puppeteer to inject a script here. You will be able to use axe-core without having to deal with Puppeteer at all.

Reply:
Ah this comment might need updated then:
https://github.com/webdriverio/webdriverio/pull/5929/files/fdeff0e094490e4cc12ecdfbcc7454856317f216#diff-e8517450f670f58b95cd042559d95b4938983d60caf1aa97b6567caaf10ceb8dR42-R43